scripts/RelayHubGasEstimation.ts

```diff
@@ -505,122 +567,101 @@ const estimateGas = async () => {
     `Gas Used: ${txReceipt.gasUsed.toString()} - Cumulative Gas Used: ${txReceipt.cumulativeGasUsed.toString()}\n`
   );
 
-  async function forgeRequest(
-    transferReceiver: string,
-    balanceToTransfer: number,
-    fees: string,
-    token: UtilToken,
-    relayRequest: RelayRequest,
-    forwarder: SmartWallet,
-  ) {
-
-    const isSponsored = fees === '0';
-    const tokenContract = isSponsored
-      ? ethers.constants.AddressZero
-      : token.address;
-
-    const encodedFunction = token.interface.encodeFunctionData(
-      'transfer',
-      [transferReceiver, balanceToTransfer]
-    );
-
-    const estimatedDestinationCallGas = await ethers.provider.estimateGas(
-      {
-        from: forwarder.address,
-        to: token.address,
-        gasPrice,
-        data: encodedFunction,
-      }
-    );
+  logTitle('Gas estimation tests for token transfer - with token payment');
+  await estimateTokenTransferGasOverhead('5000');
 
-    const internalDestinationCallCost = correctEstimatedCallCost(
-      estimatedDestinationCallGas.toNumber()
-    );
+  logTitle('Gas estimation tests for token transfer - without token payment');
+  await estimateTokenTransferGasOverhead('0');
+}
 
-    const estimatedTokenPaymentGas = await ethers.provider.estimateGas({
-      from: forwarder.address,
-      to: token.address,
-      data: token.interface.encodeFunctionData('transfer', [
-        relayWorker.address,
-        fees,
-      ]),
-    });
+async function runDeployEstimation(tokenAmount = '0') {
+  const {owner, relayHub,factory, token, verifier} = await deployAndSetup();
```

Comment:
This function also deploys the SW in without using the relayHub, and it's totally unnecessary in this case. I'd change that function to avoid it (also because the log created is quite confusing IMHO)

Reply:
Refactored.